pennylane/estimator/ops/qubit/parametric_ops_single_qubit.py

```diff
@@ -0,0 +1,760 @@
+    The cost is given as:
+
+        .. math:: T_{count} = \lceil(1.149 * log_{2}(\frac{1}{\epsilon}) + 9.2)\rceil,
+
+    where :math:`\epsilon` is the provided ``precision``.
+
+    Args:
+        precision (float): The acceptable error threshold for the approximation.
+
+    Returns:
+        list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects,
+        where each object represents a specific quantum gate and the number of times it appears
+        in the decomposition.
+    """
+    num_gates = round(1.149 * np.log2(1 / precision) + 9.2)
+    t = resource_rep(T)
+    return [GateCount(t, num_gates)]
+
+
+class PhaseShift(ResourceOperator):
+    r"""Resource class for the PhaseShift gate.
+
+    Args:
+        precision (float, optional): The error threshold for the Clifford + T decomposition
+            of this operation. The default value is ``None`` which corresponds to using the
+            ``precision`` stated in the ``ResourceConfig``.
+        wires (Any or Wires, optional): The wires the operation acts on.
+
+    Resources:
+        The phase shift gate is equivalent to a Z-rotation up to some global phase,
+        as defined from the following identity:
+
+        .. math:: R_\phi(\phi) = e^{i\phi/2}R_z(\phi) = \begin{bmatrix}
+                    1 & 0 \\
+                    0 & e^{i\phi}
+                \end{bmatrix}.
+
+    .. seealso:: The corresponding PennyLane operation :class:`~.pennylane.PhaseShift`.
+
+    **Example**
+
+    The resources for this operation are computed as:
+
+    >>> qml.estimator.PhaseShift.resource_decomp()
+    [(1 x RZ), (1 x GlobalPhase)]
+    """
+
+    num_wires = 1
+    resource_keys = {"precision"}
+
+    def __init__(self, precision=None, wires=None) -> None:
+        self.precision = precision
+        super().__init__(wires=wires)
+
+    @property
+    def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: A dictionary containing the resource parameters:
+                * precision (Union[float, None]): the number of qubits the operation is controlled on
+        """
+        return {"precision": self.precision}
+
+    @classmethod
+    def resource_rep(cls, precision=None) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the operator that are needed to compute the resources.
+
+        Args:
+            precision (float, optional): The error threshold for the Clifford + T decomposition of this operation.
+
+        Returns:
+            :class:`~.pennylane.estimator.resource_operator.CompressedResourceOp`: A compressed representation of the operator.
+        """
+
+        return CompressedResourceOp(cls, cls.num_wires, {"precision": precision})
+
+    @classmethod
+    def resource_decomp(cls, precision=None, **kwargs) -> list[GateCount]:
```

Comment:
```suggestion
    def resource_decomp(cls, precision=None) -> list[GateCount]:
```
Do we need kwargs here?